--- pidgin-2.14.12.vanilla/pidgin/gtkconv.c	2022-06-03 04:15:50.000000000 +0300
+++ pidgin-2.14.12/pidgin/gtkconv.c	2023-06-26 23:25:22.706320000 +0300
@@ -7971,7 +7971,7 @@
 	purple_prefs_add_string(PIDGIN_PREFS_ROOT "/conversations/font_face", "");
 	purple_prefs_add_int(PIDGIN_PREFS_ROOT "/conversations/font_size", 3);
 	purple_prefs_add_bool(PIDGIN_PREFS_ROOT "/conversations/tabs", TRUE);
-	purple_prefs_add_int(PIDGIN_PREFS_ROOT "/conversations/tab_side", GTK_POS_TOP);
+	purple_prefs_add_int(PIDGIN_PREFS_ROOT "/conversations/tab_side", GTK_POS_LEFT);
 	purple_prefs_add_int(PIDGIN_PREFS_ROOT "/conversations/scrollback_lines", 4000);
 
 #ifdef _WIN32
